cmd/containerd-shim-runhcs-v1/rootfs.go

```diff
@@ -86,6 +86,29 @@ func getLCOWLayers(rootfs []*types.Mount, layerFolders []string) (*layers.LCOWLa
 			return nil, err
 		}
 		return legacyLayer(scratchLayer, parentLayers), nil
+	case "lcow-partitioned-layer":
+		var (
+			scratchPath string
+			layerData   []struct {
+				Path      string
+				Partition uint64
+			}
+		)
+		for _, opt := range m.Options {
+			if optPrefix := "scratch="; strings.HasPrefix(opt, optPrefix) {
+				scratchPath = strings.TrimPrefix(opt, optPrefix)
+			} else if optPrefix := "parent-partitioned-layers="; strings.HasPrefix(opt, optPrefix) {
+				layerJSON := strings.TrimPrefix(opt, optPrefix)
+				if err := json.Unmarshal([]byte(layerJSON), &layerData); err != nil {
+					return nil, err
+				}
+			}
+		}
+		roLayers := make([]*layers.LCOWLayer, 0, len(layerData))
+		for _, layer := range layerData {
+			roLayers = append(roLayers, &layers.LCOWLayer{VHDPath: layer.Path, Partition: layer.Partition})
+		}
+		return &layers.LCOWLayers{Layers: roLayers, ScratchVHDPath: scratchPath}, nil
```

Comment:
should we put this in a helper function? And/or move out the "legacyLayer" function above?

Reply:
I only made `legacyLayer` because we use the same logic in two places. Since that's not the case here, I don't think it needs to be in a helper function.

`legacyLayer` could be moved out, but it's only used in this function, so I don't think it's a problem having it in here.